fontra2fontir/src/toir.rs

```diff
@@ -159,66 +155,59 @@ fn to_ir_glyph(
 
 #[allow(dead_code)] // TEMPORARY
 fn add_to_path<'a>(
-    glyph_name: GlyphName,
     path_builder: &'a mut GlyphPathBuilder,
     points: impl Iterator<Item = &'a FontraPoint>,
-) -> Result<(), WorkError> {
+) -> Result<(), PathConversionError> {
     // Walk through the remaining points, accumulating off-curve points until we see an on-curve
     // https://github.com/googlefonts/glyphsLib/blob/24b4d340e4c82948ba121dcfe563c1450a8e69c9/Lib/glyphsLib/pens.py#L92
     for point in points {
-        let point_type = point
-            .point_type()
-            .map_err(|e| WorkError::InvalidSourceGlyph {
-                glyph_name: glyph_name.clone(),
-                message: format!("No point type for {point:?}: {e}"),
-            })?;
+        let point_type = point.point_type().map_err(PathConversionError::Parse)?;
         // Smooth is only relevant to editors so ignore here
         match point_type {
-            PointType::OnCurve | PointType::OnCurveSmooth => path_builder
-                .curve_to((point.x, point.y))
-                .map_err(WorkError::PathConversionError)?,
-            PointType::OffCurveQuad | PointType::OffCurveCubic => path_builder
-                .offcurve((point.x, point.y))
-                .map_err(WorkError::PathConversionError)?,
+            PointType::OnCurve | PointType::OnCurveSmooth => {
+                path_builder.curve_to((point.x, point.y))?
+            }
+            PointType::OffCurveQuad | PointType::OffCurveCubic => {
+                path_builder.offcurve((point.x, point.y))?
+            }
         }
     }
     Ok(())
 }
 
-fn to_ir_path(glyph_name: GlyphName, contour: &FontraContour) -> Result<BezPath, WorkError> {
+fn to_ir_path(glyph_name: GlyphName, contour: &FontraContour) -> Result<BezPath, BadGlyph> {
     // Based on glyphs2fontir/src/toir.rs to_ir_path
     // TODO(https://github.com/googlefonts/fontc/issues/700): share code
     if contour.points.is_empty() {
         return Ok(BezPath::new());
     }
 
-    let mut path_builder = GlyphPathBuilder::new(glyph_name.clone(), contour.points.len());
+    let mut path_builder = GlyphPathBuilder::new(contour.points.len());
 
     if !contour.is_closed {
         let first = contour.points.first().unwrap();
         let first_type = first
             .point_type()
-            .map_err(|e| WorkError::InvalidSourceGlyph {
-                glyph_name: glyph_name.clone(),
-                message: format!("No point type for {first:?}: {e}"),
-            })?;
+            .map_err(|e| BadGlyph::parse(glyph_name.clone(), e))?;
         if first_type.is_off_curve() {
-            return Err(WorkError::InvalidSourceGlyph {
-                glyph_name: glyph_name.clone(),
-                message: String::from("Open path starts with off-curve points"),
-            });
+            return Err(BadGlyph::parse(
```

Comment:
I find the use of parse very weird here. Why isn't this just constructing an error?

Reply:
`parse` is a method for constructing `BadGlyph::Parse`.